docs/src/cookbooks/two-side-column.md

```diff
@@ -0,0 +1,888 @@
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class RegisterModule extends Module {
+  val io = IO(new Bundle {
+    val in  = Input(UInt(12.W))
+    val out = Output(UInt(12.W))
+  })
+
+  val registerWithInit = RegInit(42.U(12.W))
+  registerWithInit := registerWithInit - 1.U
+  io.out := io.in
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new RegisterModule)
+```
+</td>
+         </tr>
+ <tr>
+<td>
+
+```
+module RegisterModule(
+  input        clock,
+  input        reset,
+  input  [7:0] in,
+  output [7:0] out,
+  input        differentClock,
+  input        differentSyncReset,
+  input        differentAsyncReset
+);
+
+reg [7:0] registerWithoutInit;
+reg [7:0] registerWithInit;
+reg [7:0] registerOnDifferentClockAndSyncReset;
+reg [7:0] registerOnDifferentClockAndAsyncReset;
+
+
+ always @(posedge clock) begin
+    registerWithoutInit <= in + 8'h1;
+ end
+
+ always @(posedge clock) begin
+    if (reset) begin
+      registerWithInit <= 8'd42;
+    end else begin
+      registerWithInit <= registerWithInit - 8'h1;
+    end
+  end
+ 
+ always @(posedge differentClock) begin
+    if (differentSyncReset) begin
+      registerOnDifferentClockAndSyncReset <= 8'h42;
+    end else begin
+      registerOnDifferentClockAndSyncReset <= in - 8'h1;
+    end
+ end
+  
+ always @(posedge differentClock or posedge differentAsyncReset) begin
+   if (differentAsyncReset) begin
+      registerOnDifferentClockAndAsyncReset <= 8'h24;
+   end else begin
+      registerOnDifferentClockAndAsyncReset <= in + 8'h2;
+   end
+ end
+
+  assign out = in + 
+    registerWithoutInit + 
+    registerWithInit + 
+    registerOnDifferentClockAndSyncReset + 
+    registerOnDifferentClockAndAsyncReset;
+
+
+endmodule
+
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class RegisterModule2 extends Module {
+  val in  = IO(Input(UInt(8.W)))
+  val out = IO(Output(UInt(8.W)))
+
+  val differentClock = IO(Input(Clock()))
+  val differentSyncReset = IO(Input(Bool()))
+  
+  val differentAsyncReset = IO(Input(AsyncReset()))
+  
+  
+  
+  val registerWithoutInit = Reg(UInt(8.W))
+  
+  val registerWithInit = RegInit(42.U(8.W))
+  
+  registerWithoutInit := in + 1.U
+  
+  registerWithInit := registerWithInit - 1.U
+  
+  val registerOnDifferentClockAndSyncReset = withClockAndReset(differentClock, differentSyncReset) {
+    val reg = RegInit("h42".U(8.W))
+    reg
+  }
+  registerOnDifferentClockAndSyncReset := in - 1.U
+  
+  val registerOnDifferentClockAndAsyncReset = withClockAndReset(differentClock, differentAsyncReset) {
+    val reg = RegInit("h24".U(8.W))
+    reg
+  }
+  registerOnDifferentClockAndAsyncReset := in + 2.U
+  
+  out := in + 
+    registerWithoutInit + 
+    registerWithInit + 
+    registerOnDifferentClockAndSyncReset + 
+    registerOnDifferentClockAndAsyncReset
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new RegisterModule2)
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# Case Statements
```

Comment:
Something is strange with the row breaks here, look at the version on viewer and see what's wrong